ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/StreamedQueryResource.java

```diff
@@ -540,6 +549,44 @@ private EndpointResponse handleScalablePushQuery(
     return EndpointResponse.ok(queryStreamWriter);
   }
 
+  private EndpointResponse handleStreamPullQuery(
+      final ImmutableAnalysis analysis,
+      final ServiceContext serviceContext,
+      final ConfiguredStatement<Query> configured,
+      final CompletableFuture<Void> connectionClosedFuture) {
+
+    final TransientQueryMetadata transientQueryMetadata = ksqlEngine
+        .createStreamPullQuery(
+            serviceContext,
+            analysis,
+            configured,
+            false
+        );
+    localCommands.ifPresent(lc -> lc.write(transientQueryMetadata));
+
+    // Be sure to close the stream in the event of any error. It's save to close it twice,
+    // which will happen if we call close inside the block, and then again when we exit the
+    // try-with-resources
+    try (final QueryStreamWriter queryStreamWriter = new QueryStreamWriter(
+        transientQueryMetadata,
+        disconnectCheckInterval.toMillis(),
+        OBJECT_MAPPER,
+        connectionClosedFuture
+    )) {
+      ksqlEngine.waitForStreamPullQuery(
+          serviceContext,
+          analysis,
+          configured,
+          transientQueryMetadata
+      );
+      // stop the response stream before sending the successful response.
+      queryStreamWriter.close();
```

Comment:
It should always get called at the end of `QueryStreamWriter.write`.  We don't explicitly call this for push queries.  Not a huge deal, but curious why they're different.

Reply:
I'll take another look. IIRC, this is a difference between the handling of push and pull queries. Since we decided to "hack" this feature in by reusing most of the code paths for transient queries, there's no "at the end" like there is for pull queries.

If we want it to be a clean implementation, I'd have to create a whole new stack of query-handling logic.

Reply:
I think I understand why this is happening.  It's because QueryStreamWriter.write isn't called at all until after this returns, but you're wanting to complete it here.  I think this shouldn't be required.